### PR TITLE
Fixed sql security issue and added event-type creation

### PR DESCRIPTION
**Pull request by rmatsuda21**
#### Category
- [ ] Bug Fix
- [x] New Feature
- [x] Hotfix

#### What does this PR do?

Fixed mssql security vulnerability. Additionally, added function to create event types.

### CODE REVIEW

File: routes/portal/events.js [hunk 1]
```diff
@@ -290,6 +294,88 @@ router.post('/event/delete/:id', middleware.checkAuthenticated, function (
 		});
 });
 
+async function createEvent(name, id) {
+	var sqlQuery = `INSERT INTO tbl_event(name, event_type_id) VALUES ('${name}', ${id});`;
```
Comment (tsheinen) [inline routes/portal/events.js]: this is vulnerable to a sql injection

File: routes/portal/events.js [hunk 2]
```diff
@@ -119,18 +121,20 @@ router.post('/meeting/update/:id', middleware.checkAuthenticated, function (
...
+
+	sqlReq.input('name', sql.NVarChar, req.body.meetingName);
+	sqlReq.input('description', sql.NVarChar, req.body.meetingDescription);
+	sqlReq.input('link', sql.NVarChar, req.body.meetingLink);
+	sqlReq.input('time', sql.NVarChar, req.body.meetingTime);
+	sqlReq.input('day', sql.NVarChar, req.body.meetingDay);
 
-	var sqlQuery = `UPDATE tbl_meeting SET name = '${name}', description = '${description}', link = '${link}', time = '${time}', day = '${day}'
+	var sqlQuery = `UPDATE tbl_meeting SET name = @name, description = @description, link = @link, time = @time, day = @day
 				WHERE id=${req.params['id']}`;
```
Comment (tsheinen) [inline routes/portal/events.js]: this is vulnerable to a sql injection

File: routes/portal/events.js [hunk 3]
```diff
@@ -290,6 +294,88 @@ router.post('/event/delete/:id', middleware.checkAuthenticated, function (
...
+			})
+			.catch((err) => {
+				res.status(400);
+				res.send('Error!');
+			});
+		
+		const sqlReq = new sql.Request();
+
+		sqlReq.input('name', sql.NVarChar, req.body.eventTypeName);
+		var sqlQuery = `INSERT INTO tbl_event_type(id, name, img) VALUES (${maxID},@name,'')`;
```
Comment (tsheinen) [inline routes/portal/events.js]: this is vulnerable to a sql injection

**Review by tsheinen — COMMENTED** [review]
hi -- the lines i commented on all look vulnerable to sql injection and I saw a few others in the file that weren't modified. If I were you i'd ctrl+f through the file for string interpolation and double check to make sure you've caught every instance of string interpolation on a query. An attacker can do a lot of nasty things with a sql injection; up to potentially arbitrary code exec

File: routes/portal/events.js (hunk 2 above)
Comment (tsheinen) [reply routes/portal/events.js]: to clarify its line 133, the where clause -- i saw you already fixed the prev line

**Comment by cdconn00** [comment]
Thanks @tcheinen! Think I got em all, if there is any others you find open a ticket and assign @Vigasaurus

**Pull request merged**